filters/HAGFilter.cpp

```diff
@@ -229,6 +242,27 @@ void HAGFilter::filter(PointView& view)
 {
     using namespace pdal::Dimension;
 
+    // If using a raster as the DTM, check each point against it and write
+    // the HAG dimension
+    if (m_rasterName != "NONE") {
+      static std::vector<double> data;
+      for (PointId i = 0; i < view.size(); ++i)
+      {
+        double x = view.getFieldAs<double>(Id::X, i);
+        double y = view.getFieldAs<double>(Id::Y, i);
+        double z = view.getFieldAs<double>(Id::Z, i);
+
+        if (m_raster->read(x, y, data) == gdal::GDALError::None)
```

Comment:
I wonder what should be done in the "else" case.

Reply:
I expect it should do nothing in the default case. If there is no ground defined at that point then there is no HAG, right? Maybe we can add an option to allow:

- default to a configurable value
- the Z of the point